mcp/motors/actuators.c

```diff
@@ -993,10 +1003,289 @@ static void DoShutter(void)
 // ============================================================================
 // Do elevation axis lock logic: check status, determine if we are locked, etc.
 // ============================================================================
+
+/**
+ * @brief Ask the lock pin EZStepper about the limit switches.
+ * @param[in] pBus Pointer to EZStepper ezbus struct
+ * @param[in] who Char in id array for the lock pin, i.e. id[LOCKNUM]
+ * @param[out] pLims Contains status of all switch inputs
+ * @param[out] pPos Contains commanded motor position
+ * @return -1 if either read failed, 0 if both succeeded.
+ */
+static int GetLockData(struct ezbus* pBus, char who, int* pLims, int* pPos)
+{
+    int retval = 0;
+    int errorCodeLim = EZ_ERR_OK;
+    int errorCodePos = EZ_ERR_OK;
+
+    if ((errorCodeLim = EZBus_ReadInt(pBus, who, "?4", pLims) != EZ_ERR_OK)) {
+        blast_info("GetLockData: EZBus_ReadInt error -- lims, error code = %d", errorCodeLim);
+    }
+    if ((errorCodePos = EZBus_ReadInt(pBus, who, "?0", pPos) != EZ_ERR_OK)) {
+        blast_info("GetLockData: EZBus_ReadInt error -- pos, error code = %d", errorCodePos);
+    }
+    if ((errorCodeLim != EZ_ERR_OK) || (errorCodePos != EZ_ERR_OK)) {
+        retval = -1;
+    }
+    return retval;
+}
+
+/**
+ * @brief Update structs related to elevation lock.
+ * The NiC MCC does this via the BlastBus to give it a chance to know what's
+ * going on. The ICC reads it directly to get more promptly the answer (since
+ * all these fields are slow).
+ * @param nic flag to determine where the state data is queried from. 1 =
+ * other computer, 0 = in-charge computer
```

Comment:
wow i hate that this needs to take something like ``!InCharge`` as an argument rather than ``InCharge``